### PR TITLE
Do not cast to FP32 on SFT trainer

### PR DESCRIPTION
**Pull request by samsja**
# fix memory issue

This remove the float32 casting from sft. It fix memory usage and we don't need fp32 logits for sft only for RL

**Command**

```bash
uv run sft --model.name Qwen/Qwen3-0.6B --model.compile --model.liger-kernel --data.type fake --data.seq-len 2048 --data.batch-size 64 --data.micro-batch-size 1 --bench 
```

**Previous**

Screenshot 2025-09-12 at 12 02 51 PM

**Now**

Screenshot 2025-09-12 at 12 00 56 PM

### CODE REVIEW

File: src/prime_rl/trainer/sft/train.py [hunk 1]
```diff
@@ -170,7 +169,7 @@ def train(config: SFTTrainerConfig):
 
             with maybe_context_parallel:
                 # Forward pass
-                logits = forward(model, input_ids, position_ids).contiguous()
+                logits = model(input_ids=input_ids, position_ids=position_ids).logits
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/sft/train.py]: the reason we used a helper func here is to have jaxtype catch type and shape errors (remember position ids not being long). can we maybe change signature to have a boolean flag for whether to cast or not. or not cast by default and then do the cast optionally on the caller

**Review by mikasenghaas — COMMENTED** [review]
nice!! small comment then good to go

**Comment by mikasenghaas** [comment]
added back the fp32 cast on rl trainer to avoid possible regression there

**Comment by samsja** [comment]
> added back the fp32 cast on rl trainer to avoid possible regression there

oh good catch
